base/docs/basedocs.jl

```diff
@@ -2871,6 +2871,11 @@ julia> obj.x
 1
 ```
 
+One should overload `getproperty` only when necessary, as it can be confusing if 
+the behavior of the syntax `obj.f` is unusual.
+Also note that using methods is often preferable. See also this style guide documentation 
```

Comment:
```suggestion
One should overload `getproperty` only when necessary, as it can be confusing if
the behavior of the syntax `obj.f` is unusual.
Also note that using methods is often preferable. See also this style guide documentation
```